R/git-default-branch.R

```diff
@@ -93,39 +93,18 @@ NULL
 #' git_default_branch()
 #' }
 git_default_branch <- function() {
-  repo <- git_repo()
+  git_default_branch_(github_remote_config())
+}
 
-  # TODO: often when we call git_default_branch(), we already have a GitHub
-  # configuration or target repo, as produced by github_remote_config() or
-  # target_repo(). In that case, we don't need to start from scratch as we do
-  # here. But I'm not sure it's worth adding complexity to allow passing this
-  # data in.
-
-  # TODO: this critique feels somewhat mis-placed, i.e. it brings up a general
-  # concern about a repo's config (or the user's permissions and creds)
-  # related to whether github_remotes() should be as silent as it is about
-  # 404s
-  critique_remote <- function(remote) {
-    if (remote$is_configured && is.na(remote$default_branch)) {
-      ui_bullets(c(
-        "x" = "The {.val {remote$name}} remote is configured, but we can't
-               determine its default branch.",
-        " " = "Possible reasons:",
-        "*" = "The remote repo no longer exists, suggesting the local remote
-               should be deleted.",
-        "*" = "We are offline or that specific Git server is down.",
-        "*" = "You don't have the necessary permission or something is wrong
-               with your credentials."
-      ))
-    }
-  }
```

Comment:
Deleted because I think it's now more clear that it's the responsibility of `github_remote_config()` to report such problems, and `github_remote_config()` is now more consistently called before `git_default_branch()` where it matters.

Reply:
For example, if you're offline, `pr_resume()` will emit this message even though you can still checkout the local branches that it lists.